e2e/specs/results-viewer.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { expect } from '@playwright/test';
+import { type Visit } from '@openmrs/esm-framework';
+import { generateRandomPatient, type Patient, startVisit, endVisit, deletePatient } from '../commands';
+import { test } from '../core';
+import { ResultsViewerPage, VisitsPage } from '../pages';
+
+let patient: Patient;
+let visit: Visit;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+  visit = await startVisit(api, patient.uuid);
+});
+
+test('Record and edit test results', async ({ page }) => {
+  const resultsViewerPage = new ResultsViewerPage(page);
+  const visitsPage = new VisitsPage(page);
+
+  await test.step('When I visit the results viewer page', async () => {
+    await resultsViewerPage.goTo(patient.uuid);
+  });
+
+  await test.step('And I click on the `Clinical forms` button on the siderail', async () => {
+    await page.getByLabel(/clinical forms/i).click();
+  });
+
+  await test.step('Then I should see the clinical forms workspace', async () => {
+    const headerRow = resultsViewerPage.formsTable().locator('thead > tr');
+
+    await expect(page.getByPlaceholder(/search this list/i)).toBeVisible();
+    await expect(headerRow).toContainText(/form name \(a-z\)/i);
+    await expect(headerRow).toContainText(/last completed/i);
+    await expect(page.getByRole('cell', { name: /laboratory test results/i })).toBeVisible();
+  });
+
+  await test.step('When I launch the `Laboratory Test Results` form', async () => {
+    await page.getByText(/laboratory test results/i).click();
+  });
+
+  await test.step('Then it should launch in the workspace', async () => {
+    await expect(page.getByRole('button', { name: /hematology/i, exact: true })).toBeVisible();
+    await expect(page.getByRole('button', { name: 'Chemistry', exact: true })).toBeVisible();
```

Comment:
```suggestion
    await expect(page.getByRole('button', { name: /chemistry/i, exact: true })).toBeVisible();
```

Reply:
@kdaud  This won't pass because there will be more than one match. 
Below the form, there's a `Next Chemistry` button.
That's the reason I'm doing it differently here.
If there was `id`, I probably would have opted for that but there isn't.